### ISSUE TITLE
Sometimes occur memory leak error message when i get some infomation from cf

### ISSUE THREAD

**Issue by lijie-advantech** [cloudfoundry/cf-java-client#806]
when i get some information from cloud foundry , some times will occur error below, is there some bugs or i must release memory somewhere, but i don't find some release function.
LEAK: ByteBuf.release() was not called before it's garbage-collected. Enable advanced leak reporting to find out where the leak occurred. To enable advanced leak reporting, specify the JVM option '-Dio.netty.leakDetection.level=advanced' or call ResourceLeakDetector.setLevel() See http://netty.io/wiki/reference-counted-objects.html for more information.

**Comment by nebhale** [comment]
Yes, this leak (in `reactor-Netty` our underlying networking library) has been fixed in a forthcoming release.  This should no longer occur in 2.17.0 or later once it is released.  In the meantime, the leak is quite minor and should not affect your use of the library significantly.

/cc @smaldini @violetagg

**Comment by lijie-advantech** [comment]
I get information from cloud foundry every 2 minutes, after a night, i got the message like that io.netty.util.internal.OutOfDirectMemoryError: failed to allocate 16777216 byte(s) of direct memory (used: 234881031, max: 239075328). 
I think this leak affect my program. in addition, I want to know when the new release will come.

**Comment by nebhale** [comment]
The direct memory leak is a different issue and has also been fixed in a forthcoming release.  We do not set release dates, rather releasing when there is sufficient change to warrant a release.  At the moment, we’re working through an SSL issue in [another ticket](https://github.com/cloudfoundry/cf-java-client/issues/797) and once that is complete, we’ll likely release.

**Comment by nebhale** [comment]
This was fixed in the 2.17.0 release.
